rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,127 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
+        to_asset: Optional[Asset],
+    ) -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE source_type=? '
+        )
+        # Start by adding the manual source type to the list of params
+        params = [HistoricalPriceOracle.MANUAL.serialize_for_db()]  # pylint: disable=no-member
+        if from_asset is not None:
+            querystr += 'AND from_asset=? '
+            params.append(from_asset.identifier)
+        if to_asset is not None:
+            querystr += 'AND to_asset=? '
+            params.append(to_asset.identifier)
+        querystr += 'ORDER BY timestamp'
+        query = cursor.execute(querystr, tuple(params))
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in query.fetchall()
+        ]
+
+    @staticmethod
+    def edit_manual_price(entry: HistoricalPrice) -> bool:
+        """Edits a manually inserted historical price. Returns false if no row
+        was updated and true otherwise.
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'UPDATE price_history SET price=? WHERE from_asset=? AND to_asset=? '
+            'AND source_type=? AND timestamp=? '
+        )
+        entry_serialized = entry.serialize_for_db()
+        # Price that is the last entry should be the first and the rest of the
+        # positions are correct in the tuple
+        params_update = entry_serialized[-1:] + entry_serialized[:-1]
+        try:
+            cursor.execute(querystr, params_update)
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to edit manual historical prices from {entry.from_asset} '
+                f'to {entry.to_asset} at timestamp: {str(entry.timestamp)} due to {str(e)}',
+            )
+            return False
+
+        if cursor.rowcount == 1:
+            connection.commit()
+            return True
+        return False
+
+    @staticmethod
+    def delete_manual_price(
+            from_asset: 'Asset',
+            to_asset: 'Asset',
+            timestamp: Timestamp,
+    ) -> bool:
+        """
+        Deletes a manually inserted historical price given by its primary key.
+        Returns True if one row was deleted and False otherwise
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'DELETE FROM price_history WHERE from_asset=? AND to_asset=? '
+            'AND timestamp=? AND source_type=?'
+        )
+        query_list = [
```

Comment:
Why make it into a list and turn it into a tuple right afterwards? Just initialize a tuple since you don't modify it.